### DISCUSSION TITLE
Issue with JIT and Gatsby

### DISCUSSION THREAD

**Discussion by ebg-digital-marketing** [tailwindlabs/tailwindcss#5514] [Help] [answered]
Tailwind version: ^2.2.7
PostCSS version: ^8.3.6
gatsby-plugin-postcss: ^4.11.0
Framework: GatsbyJS

I have some styles that are applied to components based on props, yet those styles don't get applied correctly when using JIT mode. For example, here is a button that gets a specific class based upon the `color` prop that gets passed.

```javascript
// src/components/Button.js

import { Link } from 'gatsby';

const Button = ({ color, to, children }) => {
  return (
    <Link to={to} className={`btn btn-${color}`}>
      {children}
    </Link>
  );
};

export default Button;
```

```javascript
// src/pages/index.js

import Button from '@components/Button';

const Home = () => {
  return (
    <div className="max-w-500 mx-auto mt-152">
      <Button color="indigo" to="/about/careers">
        Sign Up
      </Button>
    </div>
  );
};

export default Home;
```

My button styles in index.css are as follows:

```css
/* src/assets/css/index.css */

@layer utilities {
  .btn {
    @apply inline-block no-underline pt-8 pb-8 px-24 rounded-md text-center transition-colors;
  }

  .btn-indigo {
    @apply text-white bg-indigo hover:bg-indigo-500;
  }
}
```
My tailwind.config.js is:

```javascript
module.exports = {
  mode: 'jit',
  purge: ['./src/**/*.js'],
  theme: {
    etc...
  }
```
When I comment out `mode: 'jit'`, the styles will be applied correctly. But as soon as I add JIT mode, the styles don't work. The `btn btn-indigo` class is still applied in the DOM element, but the actually `btn-indigo` class is not generated.

![Screen Shot 2021-09-15 at 12 34 03 PM](https://user-images.githubusercontent.com/79675893/133498282-7abf23d8-cfd8-423b-9ee9-ffadac651639.png)

**Comment by ebg-digital-marketing** [comment]
I just moved my `btn` and `btn-indigo` classes outside of the `@layer utilities {}` section, and now it is working. This seems to be a short-term workaround, but is there a way I can keep my `.btn` styles within `@layer utilities` and still have JIT working? Or perhaps it is best to just remove the `@layer` calls completely?

```css
/* src/assets/css/index.css */
@tailwind base;
@tailwind components;

.btn {
  @apply inline-block no-underline pt-8 pb-[7px] px-24 rounded-md text-center transition-colors;
}

.btn-indigo {
  @apply text-white bg-indigo hover:bg-indigo-500;
}

@tailwind utilities;

@layer utilities {
  etc...
}
```

**Comment by ebg-digital-marketing** [comment, marked as answer]
I found the answer to my question here: https://tailwindcss.com/docs/optimizing-for-production#removing-all-unused-styles

**Reply by tordans** [reply]
@rrichetto25 to me it looked like an issue of https://tailwindcss.com/docs/optimizing-for-production#writing-purgeable-html. The term "btn-indigo" cannot be dynamically created but needs to be "searchable" as a string in the code. AFAIK it would even be enough to just add a comment in your component `// color=indigo uses btn-indigo`.
